### DISCUSSION TITLE
How to change gravity?

### DISCUSSION THREAD

**Discussion by fabkury** [Hartrik/sand-game-js#3] [General]
I wanted to make some animated GIFs using this tool, if that's ok.

Imagine for example an hourglass. I wanted to be able to turn it upside down.

I can easily turn an _image_ upside down, outside of the tool (e.g. in GIMP), but would there be an easy way to change the direction of gravity in the simulation?

If I can have full control of gravity, I can rotate the images "in post production" in sync with the change in the direction of gravity, hence simulating a real rotation of an hourglass.

I am not experienced with Javascript, but I code in other languages. I can try to implement the gravity chanmge feature in a rough-but-works way if needed.

**Comment by Hartrik** [comment]
Yes, why not :)

However, the gravity direction is somewhat hardcoded in the simulation, and there isn't a straightforward way to change it. A simpler approach might be to mirror flip the content of the canvas, which could simulate the effect you're looking for

**Reply by fabkury** [reply]
Thanks :)
I see what you're saying. Would you mind giving me a general direction to go, if I am to implement that myself?

**Reply by Hartrik** [reply]
You can add a flip button into dev/test tools. See how _Fill_ is implemented in `gui/ComponentViewTestTools`. Re-introduce [this](https://github.com/Hartrik/sand-game-js/commit/64b804af01e9d42dc996974c4f1bdb4f782c3742#diff-cc6e32d9aec9301c14c788cd8f1da4ecbe28f222b04c572a6e1225b8786d73e1L29-L45) method that I recently deleted. Create an action similar to `gui/ActionFill` – use `sandGame` and `sandGame.graphics()` ...

If you do it this way and want to create a pull request, I will approve it
